src/lib.rs

```diff
@@ -4,32 +4,40 @@
 /// type.
 #[macro_export]
 macro_rules! serde_visitor {
-    ($name:ident, $visitor:ident) => (
+    ($name:ident, $visitor:ident, $($variant:ident),+) => (
+        #[allow(non_upper_case_globals)]
+        const $name: &'static [&'static str] = &[$(stringify!($variant)),+];
+
```

Comment:
I was surprised as well... It surely is a workaround and afaik the rust core team is aware of this flaw as well. As soon as there is a better way it should be changed.

I will move it into visit_str as you suggested.